projects/index.mjs

```diff
@@ -0,0 +1,105 @@
+/** Route handler for the /projects endpoint */
+import express from 'express'
+import * as logic from './projects.mjs'
+import * as utils from '../utilities/shared.mjs'
+import cors from 'cors'
+import auth0Middleware from '../auth/index.mjs'
+import 'dotenv/config'
+
+let router = express.Router()
+router.use(
+  cors({
+    methods: 'GET',
+    allowedHeaders: [
+      'Content-Type',
+      'Content-Length',
+      'Allow',
+      'Authorization',
+      'Location',
+      'ETag',
+      'Connection',
+      'Keep-Alive',
+      'Date',
+      'Cache-Control',
+      'Last-Modified',
+      'Link',
+      'X-HTTP-Method-Override',
+    ],
+    exposedHeaders: '*',
+    origin: '*',
+    maxAge: '600',
+  })
+)
+
+/**
+ * Validates queries for /projects endpoint.
+ * 
+ * @param queries A JSON object of query parameters for the project list lookup.
+ * @return A boolean indicating whether validations passed or failed.
+ */
+function validateQueries(queries) {
+
+}
+
+/**
+ * Authenticated endpoint to return a list of a user's projects.
+ * 
+ * @param options A JSON object of options for the project list lookup.
+ * @return An unexpanded list of the user's projects. Returns empty list if none found.
+ */
+export async function respondWithProjects(user, options, res){
+  // Set option defaults
+  let hasRoles       = options.hasRoles       ?? 'ALL' 
+  let exceptRoles    = options.exceptRoles    ?? 'NONE'
+  let createdBefore  = options.createdBefore  ?? 'NOW'
+  let modifiedBefore = options.modifiedBefore ?? 'NOW'
+  let createdAfter   = options.createdAfter   ?? 0
+  let modifiedAfter  = options.modifiedAfter  ?? 0
+  let fields         = options.fields         ?? ['id', 'title']
+  let count          = options.count          ?? false
+  let {isPublic, hasCollaborators, tags} = options
+
+  let projects = await logic.getUserProjects(user)
+
+  if (hasRoles !== 'ALL') {
+    projects = projects.filter(project => (
+      project.roles.some(role => hasRoles.includes(role))
+    ))
+  }
+  if (exceptRoles !== 'NONE') {
+    projects = projects.filter(project => {
+      !project.roles.some(role => exceptRoles.includes(role))
+    })
+  }
+
+  if (createdBefore === 'NOW') {
+    createdBefore = Date.now()
+  }
+  projects = projects.filter(project => createdAfter < project.created && project.crated < createdBefore)
+
+  if (modifiedBefore === 'NOW') {
+    modifiedBefore = Date.now()
+  }
+  projects = projects.filter(project => modifiedAfter < project.lastModified && project.lastModified < modifiedAfter)
+
+  if (count) {
```

Comment:
For sematics, I would prefer that if there is a count, return something that looks like a count. Having two `res.send()` that send different types is clearer and easier to manage. Here is the chance to change headers for the type of response as well. `send()` just pushes back the body as text (number here) and `json()` sends back the array as JSON.